pygit2/repository.py

```diff
@@ -56,8 +56,10 @@
 
 class Repository(_Repository):
 
-    def __init__(self, *args, **kwargs):
-        super(Repository, self).__init__(*args, **kwargs)
+    def __init__(self, path, *args, **kwargs):
+        if not isinstance(path, str):
```

Comment:
`six` wasn't a dependency of this project so I was hesitant to include it.

It will actually not break though. On py2, you'll pass a `str` or `unicode` to this function. `unicode` has `.decode` and will decode to `str`. py2 `str` is accepted by Repository just fine. On py3 you'll pass a `str` or `bytes` to this function. `bytes` will `.decode` to `str`. py3 `str` is also accepted by `Repository`.

If you pass anything else then it'll break with a vague error about not having `.decode`, but this is the best I could come up with without `six`.